pulp_smash/tests/rpm/api_v2/test_schedule_consumer.py

```diff
@@ -0,0 +1,171 @@
+    RPM_UNSIGNED_FEED_URL,
+)
+from pulp_smash.tests.rpm.api_v2.utils import (
+    gen_distributor,
+    gen_repo,
+)
+from pulp_smash.tests.rpm.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+REQUIRED_SERVICES = ('goferd',)
+
+REQUIRED_PACKAGES = frozenset(('pulp-consumer-client',
+                               'pulp-rpm-consumer-extensions',
+                               'pulp-agent',
+                               'pulp-rpm-handlers',
+                               'pulp-rpm-yumplugins',
+                               'python-gofer-qpid',))
+
+
+class ScheduledTaskConsumer(unittest.TestCase):
+    """Verify scheduled RPM installation and un-installation work.
+
+    Test the installation and un-installation of scheduled package
+    in a consumer.
+
+    This test case targets `Pulp Smash #611
+    <https://github.com/PulpQE/pulp-smash/issues/611>`_
+    """
+
+    def setUp(self):
+        """Install required packages and start ``goferd`` service.
+
+        In order to communicate with ``pulp-consumer-client`` the service
+        ``goferd`` has to be started first.
+
+        Install required packages to be able start commands using
+        ``pulp-consumer``.
+        """
+        self.cfg = config.get_config()
+        if selectors.bug_is_untestable(2680, self.cfg.version):
+            self.skipTest('https://pulp.plan.io/issues/2680')
+
+        self.pkg_mgr = cli.PackageManager(self.cfg)
+        for package in REQUIRED_PACKAGES:
+            self.pkg_mgr.install(package)
+
+        system = config.PulpSystem(
+            self.cfg.systems[0].hostname,
+            {'shell': self.cfg.systems[0].roles['shell']}
+        )
+        self.svc_mgr = cli.ServiceManager(self.cfg, pulp_system=system)
+        self.svc_mgr.start(REQUIRED_SERVICES)
+
+        self.sudo = '' if utils.is_root(self.cfg) else ('sudo',)
+
+    def tearDown(self):
+        """Remove registered consumer and stop ``goferd`` service."""
+        self.svc_mgr.stop(REQUIRED_SERVICES)
+
+        cmd = (
+            '{} pulp-consumer -p {} -u {} unregister --force'
+            .format(self.sudo, *self.cfg.pulp_auth)
+        )
+
+        cli_client = cli.Client(self.cfg, cli.echo_handler)
+        cli_client.run((cmd.split()))
+
+    def test_all(self):
+        """Verify scheduled RPM installation and un-installation work.
+
+        Do the following:
+
+        1 - Create a consumer, and register it.
+        2 - Create, sync and publish a repo.
+        3 - Bind the consumer to the created repo.
+        4 - Schedule the installation of RPM package.
+        5 - Schedule the un-installation of the installed package.
+        """
+        cli_client = cli.Client(self.cfg, cli.echo_handler)
+
+        # Create and register a consumer.
+        consumer_id = utils.uuid4()
+        cmd = (
+            '{} pulp-consumer -u {} -p {} register --consumer-id {}'
+            .format(self.sudo, *self.cfg.pulp_auth, consumer_id)
```

Comment:
This is a syntax error under Python 3.4.

Reply:
@Ichimonji10 thanks, I will update accordingly.